lib/request-handler.js

```diff
@@ -78,8 +77,8 @@ RequestHandler.prototype.iterateThroughHosts = function (iterator, callback) {
       return callback(new errors.NoHostAvailableError(this.triedHosts));
     }
     host = item.value;
-    // set the distance relative to the client first, using the load balancing policy
-    var distance = host.getDistance(this.loadBalancingPolicy);
+    // set the distance relative to the client first
+    var distance = this.profileManager.getDistance(host);
```

Comment:
`this.loadBalancingPolicy` will be used for the query plan, the `distance()` is to determine the pool size. We should not change the distance in short periods of time, that is why the "closest" distance wins.